### DISCUSSION TITLE
Accessing Request Initiator (as available in DevTools 'Network' tab)

### DISCUSSION THREAD

**Discussion by freb** [go-rod/rod#845] [Q&A] [answered]
I am trying to track down all the scripts that are running on a page as well as which resource loaded them. All of this information seems to be in the browser Developer tools. The 'Sources' tab lists all scripts that have run. From there you can identify the network request for the resource and on the network tab you can see the initiator:

![image](https://user-images.githubusercontent.com/1723508/226816969-ee6a2773-2171-4767-8d60-6462bf347286.png)

I see a [Network.Initiator](https://chromedevtools.github.io/devtools-protocol/tot/Network/#type-Initiator) type in the protocol spec but I can't find reference to it in any of the calls or in any of my other searching.

Any idea what call returns the initiator or any other thoughts about how to get at the data I'm looking for?

**Comment by ysmood** [comment, marked as answer]
https://github.com/go-rod/rod/blob/4bbe74c3cb6b0b0a8a581168d638ae3482ff017b/page.go#L456-L458

**Reply by freb** [reply]
That pointed me in the right direction. I ended up using page.EachEvent because [Network.requestWillBeSent](https://vanilla.aslushnikov.com/?Network.requestWillBeSent) contains the initiator. Thanks!
